purchases/src/defaults/kotlin/com/revenuecat/purchases/coroutinesExtensions.kt

```diff
@@ -94,3 +94,30 @@ suspend fun Purchases.awaitSyncPurchases(): CustomerInfo {
         )
     }
 }
+
+/**
+ * Syncs subscriber attributes and then fetches the configured offerings for this user. This method is intended to
+ * be called when using Targeting Rules with Custom Attributes. Any subscriber attributes should be set before
+ * calling this method to ensure the returned offerings are applied with the latest subscriber attributes.
+ *
+ * * This method is rate limited to once per minute.
+ *
+ * Refer to [the guide](https://www.revenuecat.com/docs/tools/targeting) for more targeting information
+ * For more offerings information, see [getOfferings]
+ *
+ * Coroutine friendly version of [Purchases.syncAttributesAndOfferingsIfNeeded].
+ *
+ * @throws [PurchasesException] with the first [PurchasesError] if there's an error syncing attributes
+ * or fetching offerings.
+ * @returns The [Offerings] fetched after syncing attributes.
+ */
+@JvmSynthetic
+@Throws(PurchasesException::class)
+suspend fun Purchases.awaitSyncAttributesAndOfferingsIfNeeded(): Offerings? {
```

Comment:
Hmm should the return type be optional?

Reply:
Ohhhh, good point! This should not 😅

Reply:
Hmm this is still returning a nullable value?